hashintel/hash
# Introduce `GraphAPIError` class in HASH App backend

## 🌟 What is the purpose of this PR?




This PR introduces a `GraphApiError` class in the HASH App that is thrown instead of the standard `AxiosError` class when an error is returned from the Graph API.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1201095311341924/1202945471553579/f) _(internal)_

## 🔍 What does this change?




See commits.

## 📜 Does this require a change to the docs?



Not that I'm aware of.

## 🛡 What tests cover this?



None, the error was manually tested.

## ❓ How to test this?



1. Run the HASH App locally
2. Log in to the FE, and go to the GQL playground at `localhost:5001/playground`
3. Try to throw a GraphAPI error (for example by creating an entity type with a missing field in the schema)
4. Observe that a Graph API error is presented, rather than an `AxiosError`

## 📹 Demo



<img width="587" alt="image" src="https://user-images.githubusercontent.com/42802102/208665047-9f74fb58-b9d8-451f-8833-0c5b3c077425.png">
